### PR TITLE
interface styles: fix HD Healthbar

### LINKED ISSUE #9096: Interface Styles -> High Detail Health bars
Please check if your issue is not a duplicate by [searching existing issues](https://github.com/runelite/runelite/search?type=Issues)
**No previous described issue**

**Describe the bug**
After the latest runelite update High Detail health bars no longer show up. The old healthbars show even though the box is ticked and the plugin is activated.

**To Reproduce**
1. Go to 'Interface Styles'
2. Click on 'High Detail Health Bars'
3. See error High Details health bars not showing

**Expected behavior**
The High Detail Health Bar should be showing whenever someone is in combat.

**Screenshots**

**Environment (please complete the following information):**
 - OS: Windows 10 64bit
 - RuneLite version: 1.5.26
 - Launcher version: latest version

**Additional context**
Add any other context about the problem here, e.g. logs. Your client logs can usually be found in
`C:\Users\<your_user_name>\.runelite\logs` on Windows and `~/.runelite/logs` on Linux and macOS.

### PR DESCRIPTION
**Pull request by Hydrox6**
Fixes #9096
Requires !70

Jagex changed how the default healthbars work, and they now use sprites

This PR only uses the existing 30px and 90px HD Healthbars we have, though it can support all the other ones (50, 60, 80, 100, 120, 140, 160), and also allows easy support for the other sprite healthbars, such as the ones found in the Chambers of Xeric

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/interfacestyles/HealthbarOverride.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.interfacestyles;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import static net.runelite.api.SpriteID.HEALTHBAR_DEFAULT_BACK_100PX;
```
Comment (abextm) [inline runelite-client/src/main/java/net/runelite/client/plugins/interfacestyles/HealthbarOverride.java]: you can use a star import here

File: runelite-client/src/main/java/net/runelite/client/plugins/interfacestyles/HealthbarOverride.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import static net.runelite.api.SpriteID.HEALTHBAR_DEFAULT_FRONT_160PX;
+import static net.runelite.api.SpriteID.HEALTHBAR_DEFAULT_FRONT_30PX;
+import static net.runelite.api.SpriteID.HEALTHBAR_DEFAULT_FRONT_50PX;
+import static net.runelite.api.SpriteID.HEALTHBAR_DEFAULT_FRONT_60PX;
+import static net.runelite.api.SpriteID.HEALTHBAR_DEFAULT_FRONT_80PX;
+import net.runelite.client.game.SpriteOverride;
+import java.util.HashMap;
+
+@RequiredArgsConstructor
+public enum HealthbarOverride implements SpriteOverride
```
Comment (abextm) [inline runelite-client/src/main/java/net/runelite/client/plugins/interfacestyles/HealthbarOverride.java]:
```suggestion
enum HealthbarOverride implements SpriteOverride
```

File: runelite-client/src/main/java/net/runelite/client/plugins/interfacestyles/HealthbarOverride.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+	@Getter
+	private final int spriteId;
+
+	private final String fileName;
+
+	@Getter
+	private int padding = 1;
+
+	private static final HashMap<Integer, HealthbarOverride> MAP = new HashMap<>();
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/interfacestyles/HealthbarOverride.java]: Use `ImmutableMap`

File: runescape-api/src/main/java/net/runelite/rs/api/RSHealthBar.java [hunk 4]
```diff
@@ -32,6 +32,12 @@
 	@Import("healthScale")
 	int getHealthScale();
 
+	@Import("healthBarBackSpriteId")
```
Comment (Adam-) [inline runescape-api/src/main/java/net/runelite/rs/api/RSHealthBar.java]: `@Override`

File: runescape-api/src/main/java/net/runelite/rs/api/RSHealthBar.java [hunk 5]
```diff
@@ -32,6 +32,12 @@
...
+	int getHealthBarBackSpriteId();
+
+	@Import("healthBarFrontSpriteId")
```
Comment (Adam-) [inline runescape-api/src/main/java/net/runelite/rs/api/RSHealthBar.java]: `@Override`

**Pull request merged**